src/pkg/path/resource_path.go

```diff
@@ -201,13 +201,24 @@ func (rp dataLayerResourcePath) lastFolderIdx() int {
 }
 
 // Folder returns the folder segment embedded in the dataLayerResourcePath.
-func (rp dataLayerResourcePath) Folder() string {
+func (rp dataLayerResourcePath) Folder(escape bool) string {
 	endIdx := rp.lastFolderIdx()
 	if endIdx == 4 {
 		return ""
 	}
 
-	return rp.Builder.join(4, endIdx)
+	fs := rp.Folders()
+
+	if !escape {
+		return join(fs)
+	}
+
+	for i := range fs {
+		f := fs[i]
+		fs[i] = escapeElement(f)
+	}
+
+	return join(fs)
```

Comment:
since we know we need it escaped you could just make a builder and call `String()` if you want
```suggestion
	return Builder{}.Append(folders...).String()
```

Reply:
Huh, I guess I could.